### PR TITLE
CLI-685: Set up email:info command

### PR DESCRIPTION
**Pull request by klaw772**
**Motivation**
Supplements bulk Platform Email functionality/support by showing users the verification statuses of email domains registered to a subscription, as well as associated (or dissociated) domains for each application of the subscription.

**Proposed changes**
Add a new `email:info` command

**Alternatives considered**

**Testing steps**

1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. `./bin/acli email:info`
4. Verify that CSV files are successfully written to a folder called "domain-info-*" in the directory where `email:info` is being run.

**Merge requirements**
- [x] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by grasmash** [comment]
🤩

File: src/Command/Email/EmailInfoForSubscriptionCommand.php [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  }
+
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
```
Comment (grasmash) [inline src/Command/Email/EmailInfoForSubscriptionCommand.php:40]: This method is too big. Break it up into smaller methods.

File: src/Command/Email/EmailInfoForSubscriptionCommand.php [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      $this->io->newLine();
+
+      $failed_domains_table->render();
+      $this->io->newLine();
+
+      $applications_resource = new Applications($client);
+      $applications = $applications_resource->getAll();
+      $subscription_applications = [];
+
+      foreach ($applications as $application) {
```
Comment (grasmash) [inline src/Command/Email/EmailInfoForSubscriptionCommand.php:99]: E.g, this could be validateApplicationsExist()

File: .gitignore [hunk 3]
```diff
@@ -28,4 +28,6 @@ tests/fixtures/project/.acquia-cli.yml
 
 cx-api-spec
 dns-records.yaml
-dns-records.json
\ No newline at end of file
+dns-records.json
```
Comment (anavarre) [inline .gitignore:31]: I'd recommend adding a comment to indicate `dns-records.json` and `subscription-*` is for Platform email specifically. Would prevent `git blame` a year from now when we don't recall why it was added.

File: src/Command/Email/EmailInfoForSubscriptionCommand.php [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+  /**
+   * Renders tables showing email domain verification statuses,
+   * as well as exports these statuses to respective CSV files.
+   *
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   * @param \AcquiaCloudApi\Response\SubscriptionResponse $subscription
+   * @param array $domain_list
+   *
+   * @return void
+   */
```
Comment (anavarre) [inline src/Command/Email/EmailInfoForSubscriptionCommand.php:83]: Misses `@throws`. See if you can import `\League\Csv\CannotInsertRecord`.

File: src/Command/Email/EmailInfoForSubscriptionCommand.php [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+  /**
+   * Renders a table of applications in a subscription and the email domains
+   * associated or dissociated with each application.
+   *
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   * @param \AcquiaCloudApi\Connector\Client $client
+   * @param $subscription_applications
+   *
+   * @return void
+   */
```
Comment (anavarre) [inline src/Command/Email/EmailInfoForSubscriptionCommand.php:228]: Ditto.

File: src/Command/Email/EmailInfoForSubscriptionCommand.php [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+
+  /**
+   * Returns a human-readable string of whether a status code represents
+   * a failed, pending, or successful domain verification.
+   *
+   * @param string $code
+   *
+   * @return string
+   */
+  protected function showHumanReadableStatus(string $code) {
```
Comment (anavarre) [inline src/Command/Email/EmailInfoForSubscriptionCommand.php:352]: Missing return type.

File: symfony.lock [hunk 7]
```diff
@@ -86,6 +86,9 @@
     "kevinrob/guzzle-cache-middleware": {
         "version": "v3.3.1"
     },
+    "league/csv": {
+        "version": "9.8.0"
```
Comment (anavarre) [inline symfony.lock:90]: Do you need to pin to this exact version?

Comment (anavarre) [reply symfony.lock:90]: I guess that's fine, but I was wondering why you would not use `^` here. Dane or Matt can chime in and advise, I guess 😅

Comment (klaw772) [reply symfony.lock:90]: Ahhh I see - I'm happy to put in a `^` there, whatever would work best!

Comment (grasmash) [reply symfony.lock]: Yes. Use `^`.

File: src/Command/Email/EmailInfoForSubscriptionCommand.php [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+      if (!isset($subscription_applications)) {
+        return 1;
+      }
+
+      $this->renderApplicationAssociations($output, $client, $subscription, $subscription_applications);
+
+      $this->io->info("CSV files with these tables have been exported to ./subscription-{$subscription->uuid}-domains. A detailed breakdown of each domain's DNS records has been exported there as well.");
+    }
+    else {
+      $this->io->info("No email domains registered in {$subscription->name}.");
```
Comment (grasmash) [inline src/Command/Email/EmailInfoForSubscriptionCommand.php:68]:
```suggestion
      $this->io->info("No email domains registered in <options=bold>{$subscription->name}</>.");
```

File: symfony.lock [hunk 9]
```diff
@@ -87,7 +87,7 @@
         "version": "v3.3.1"
     },
     "league/csv": {
-        "version": "9.8.0"
+        "version": "^9.8.0"
```
Comment (danepowell) [inline symfony.lock:90]: @klaw772 what's the purpose of this? This breaks any subsequent `composer install` with an error about the caret not being valid.

I think `symfony.lock` is like `composer.lock`, you should never edit it manually.

Comment (klaw772) [reply symfony.lock:90]: I had added the caret based on https://github.com/acquia/cli/pull/842#discussion_r824110292 - I can add a quick hotfix to remove it if it's causing a validation error. The `composer.json` includes the caret and this was autogenerated without it, so I can remove.

cc @grasmash

Comment (klaw772) [reply symfony.lock:90]: Ah wait, I see https://github.com/acquia/cli/pull/845 now - thank you!

Comment (anavarre) [reply symfony.lock:90]: FTR this is my fault. I was sure I had reviewed `composer.json`, not `symfony.lock`.

**Pull request merged**